reporters/influx/influx.go

```diff
@@ -0,0 +1,150 @@
+package influx
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	client "github.com/influxdata/influxdb/client/v2"
+	"gopkg.in/vinxi/metrics.v0"
+)
+
+// Config stores the InfluxDB connection params.
+type Config struct {
+	URL      string
+	Database string
+	Username string
+	Password string
+	Tags     map[string]string
+}
+
+// Reporter implements an InfluxDB metrics reporter who send data to a InfluxDB server via HTTP.
+type Reporter struct {
+	config Config
+	client client.Client
+}
+
+// New creates a new InfluxDB reporter which will post the metrics to the specified server.
+func New(c Config) *Reporter {
+	re := &reporter{config: c}
+	if err := re.makeClient(); err != nil {
+		log.Printf("unable to make InfluxDB client. err=%v", err)
+	}
+	return re
+}
+
+// Report implements the metrisc.Reporter interface.
+func (r *Reporter) Report(re metrics.Report) {
+	if err := r.send(re); err != nil {
+		log.Printf("infludb: error sending metrics err=%v", err)
+	}
+}
+
+func (r *Reporter) makeClient() (err error) {
+	r.client, err = client.NewHTTPClient(client.HTTPConfig{
```

Comment:
Basically, convenience and to avoid explicit local variable declarations and boilerplate, especially in multiple assignment. E.g:
```go
func (r *Reporter) makeClient() (err error) {
	r.client, err = client.NewHTTPClient(client.HTTPConfig{
		Addr:     r.config.URL,
		Username: r.config.Username,
		Password: r.config.Password,
	})
	return
}
```

Versus:
```go
func (r *Reporter) makeClient() (err error) {
	 var err error
         r.client, err = client.NewHTTPClient(client.HTTPConfig{
		Addr:     r.config.URL,
		Username: r.config.Username,
		Password: r.config.Password,
	})
	return err
}
```

Versus:
```go
func (r *Reporter) makeClient() (err error) {
	 client, err := client.NewHTTPClient(client.HTTPConfig{
		Addr:     r.config.URL,
		Username: r.config.Username,
		Password: r.config.Password,
	})
        r.client = client
	return err
}
```